source/base/index_set.cc

```diff
@@ -63,16 +63,32 @@ IndexSet::IndexSet (const Epetra_Map &map)
 IndexSet::IndexSet (const Epetra_Map &map)
   :
   is_compressed (true),
-  index_space_size (map.NumGlobalElements()),
+#ifndef DEAL_II_WITH_64BIT_INDICES
+  index_space_size (1+map.MaxAllGID()),
+#else
+  index_space_size (1+map.MaxAllGID64()),
+#endif
   largest_range (numbers::invalid_unsigned_int)
 {
+#ifndef DEAL_II_WITH_64BIT_INDICES
+  Assert(
+    map.MinAllGID() == 0,
+    ExcMessage("The Epetra_Map does not contain the global index 0, which "
+               "means some entries are not present on any processor."));
+#else
+  Assert(
+    map.MinAllGID64() == 0,
+    ExcMessage("The Epetra_Map does not contain the global index 0, which "
+               "means some entries are not present on any processor."));
```

Comment:
make your life easier and instead of duplicating everything do it as
```
  Assert (
#if ...
               map.MinAllGID()
#else
               map.MinAllGID64()
#endif
              == 0,
             ExcMessage (...));
```